sys/auto_init/saul/auto_init_pulse_counter.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/*
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization for PULSE_COUNTER devices
+ *
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeley.edu>
+ *
+ * @}
+ */
+
+#ifdef MODULE_PULSE_COUNTER
+
+#include "log.h"
+#include "saul_reg.h"
+#include "pulse_counter_params.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define PULSE_COUNTER_NUM    (sizeof(pulse_counter_params)/sizeof(pulse_counter_params[0]))
+
+/**
+ * @brief   Allocate memory for the device descriptors
+ */
+static pulse_counter_t pulse_counter_devs[PULSE_COUNTER_NUM];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[PULSE_COUNTER_NUM];
+
+/**
+ * @brief   Reference the driver struct
+ * @{
+ */
+extern saul_driver_t pulse_counter_saul_driver;
+/** @} */
+
+
+void auto_init_pulse_counter(void)
+{
```

Comment:
And add an assert at the beginning of this function:
```c
assert(PULSE_COUNTER_NUM  == PULSE_COUNTER_INFO_NUM)
```

Reply:
addressed